### PR TITLE
[pvr] fix stuttering indruduced by 9c3adb5a9f390ad73c52469ed79de297da…

### PR DESCRIPTION
**Pull request by FernetMenta**
…d68d80

holding the lock blocks main thread while updating gui.

### CODE REVIEW

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
confirmed by user: http://forum.kodi.tv/showthread.php?tid=232630&pid=2059838#pid2059838

**Comment by Jalle19** [comment]
Could this potentially fix all the random audio stutters people have been complaining about?

File: xbmc/pvr/PVRGUIInfo.cpp [hunk 1]
```diff
@@ -747,7 +747,14 @@ void CPVRGUIInfo::UpdateBackendCache(void)
 
   // Update the backend information for all backends once per iteration
   if (m_iCurrentActiveClient == 0)
-    m_backendProperties = g_PVRClients->GetBackendProperties();
+  {
+    std::vector<SBackend> backendProperties;
+    {
+      CSingleExit exit(m_critSection);
```
Comment (Jalle19) [inline xbmc/pvr/PVRGUIInfo.cpp]: Just a quick question, is this the same as using `lock.Leave()` followed by `lock.Enter()`?

**Comment by Jalle19** [comment]
Backport at https://github.com/xbmc/xbmc/pull/7593

**Comment by FernetMenta** [comment]
@Jalle19 could be the case. and this one may also help: https://github.com/xbmc/xbmc/pull/7576

**Pull request merged**